src/components/layout/UserPortalUpdatePrompts.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @author aramsey
+ *
+ * The components for prompting users to update their user account information
+ * within the User Portal if necessary.
+ *
+ * The Snackbar is used when the user is within the grace period to update
+ * their information.
+ *
+ * The dialog is used when the grace period has expired and the user must
+ * update their information in order to use CyVerse services.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import { build } from "@cyverse-de/ui-lib";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    IconButton,
+    Snackbar,
+    Typography,
+    useTheme,
+} from "@material-ui/core";
+import { Close } from "@material-ui/icons";
+import { Alert } from "@material-ui/lab";
+
+import { useUserProfile } from "contexts/userProfile";
+import { Trans, useTranslation } from "i18n";
+import ids from "./ids";
+import { usePortalStatus } from "serviceFacades/users";
+
+function UserPortalUpdatePrompts() {
+    const [userProfile] = useUserProfile();
+    const { t } = useTranslation("common");
+    const theme = useTheme();
+
+    const baseId = ids.USER_PORTAL_UPDATE_DLG;
+    const dialogTitleId = build(baseId, ids.DIALOG_TITLE);
+
+    const [showSnackbar, setShowSnackbar] = useState(false);
+
+    const { data: userPortalResp } = usePortalStatus(userProfile?.id, (err) => {
+        console.log("Received error response from user portal API", err);
+    });
+
+    useEffect(() => {
+        setShowSnackbar(userPortalResp?.warning_required);
+    }, [userPortalResp]);
+
+    const onCloseAnnouncer = () => setShowSnackbar(false);
+
+    const onUpdateNow = () => {
+        const updateUrl = userPortalResp?.update_url;
+        if (updateUrl) {
+            const portalUrl = `${updateUrl}&redirectUrl=${encodeURIComponent(
+                window.location.href
+            )}`;
+            window.location.replace(portalUrl);
+        }
+    };
+
+    return (
+        <>
+            <Snackbar
```

Comment:
I created my own Snackbar here instead of using the CyVerse announcer because it seemed different enough:
1) We didn't want it to be auto-dismissed after a short period of time - the current CyVerse Announcer has the ability to set a custom duration but by default it has a duration set if one is not provided. In order to display the announcer infinitely, you have to omit the duration prop which wouldn't be possible with the CyVerse announcer since it sets a default.
2) I noticed it was possible to dismiss regular announcers by just clicking around the DE, so it was possible to miss the announcer entirely if you were clicking around while the announcer was just popping up